### PR TITLE
Hide "+" icon for enabled x-data

### LINKED ISSUE #7312: Hide "+" icon for enabled x-data
Once x-data has been enabled for a content, hide the "+" icon as it's confusing with both "+" and "x" icons.

image

### PR DESCRIPTION
**Pull request by ashklianko**

### CODE REVIEW

**Comment by ashklianko** [comment]
Requires https://github.com/enonic/lib-admin-ui/pull/3518 and both pulls need to remove commits updating versions

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import {PanelStripHeader, PanelStripHeaderOptions} from '@enonic/lib-admin-ui/ui/panel/PanelStripHeader';
+import {Tooltip} from '@enonic/lib-admin-ui/ui/Tooltip';
+import {i18n} from '@enonic/lib-admin-ui/util/Messages';
+import {ButtonEl} from '@enonic/lib-admin-ui/dom/ButtonEl';
+
+export interface ContentPanelStripHeaderOptions extends PanelStripHeaderOptions {
+    isOptional?: boolean;
+}
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts]:
```suggestion
export interface ContentPanelStripHeaderConfig extends PanelStripHeaderConfig {
    optional?: boolean;
}
```

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+export class ContentPanelStripHeader extends PanelStripHeader {
+
+    protected options: ContentPanelStripHeaderOptions;
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts]:
```suggestion
    protected config: ContentPanelStripHeaderConfig;
```

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    private toggler: ButtonEl;
+
+    private tooltip: Tooltip;
+
+    private enableChangedListeners: ((value: boolean) => void)[] = [];
+
+    constructor(options: ContentPanelStripHeaderOptions) {
+        super(options);
+    }
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts]:
```suggestion
    constructor(config: ContentPanelStripHeaderConfig) {
        super(config);
    }
```

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    protected initElements(): void {
+        super.initElements();
+
+        if (this.options.isOptional) {
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts]:
```suggestion
        if (this.config.optional) {
```

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            this.tooltip = new Tooltip(this.toggler, '', 200).setMode(Tooltip.MODE_GLOBAL_STATIC);
+
+            this.setTogglerState(false);
+        }
+    }
+
+    protected initListeners(): void {
+        super.initListeners();
+
+        if (this.options.isOptional) {
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentPanelStripHeader.ts]:
```suggestion
        if (this.config.optional) {
```

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardStepsPanel.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        return super.doRender().then((rendered) => {
+            this.addClass('content-wizard-steps-panel');
+
+            return rendered;
+        });
+    }
+
+    private isWizardStepFormOptional(wizardStepForm: WizardStepForm): boolean {
+        return wizardStepForm instanceof XDataWizardStepForm && wizardStepForm.isExpandable();
+    }
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardStepsPanel.ts]:
```suggestion
    private isXDataStepFormOptional(wizardStepForm: WizardStepForm): boolean {
        return wizardStepForm instanceof XDataWizardStepForm && wizardStepForm.isOptional();
    }
```

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardStepsPanel.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            panelStripHeader.onEnableChanged((state) => {
+                (wizardStepForm as XDataWizardStepForm).setExpandState(state);
+            });
+        }
+
+        return panelStripHeader;
+    }
+
+    insertNavigablePanel(item: TabBarItem, wizardStepForm: WizardStepForm, header: string, index: number, select?: boolean): number {
+        if (this.isWizardStepFormOptional(wizardStepForm)) {
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardStepsPanel.ts]:
```suggestion
        if (this.isXDataStepFormOptional(wizardStepForm)) {
```

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardStepsPanel.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import {ContentPanelStripHeader} from './ContentPanelStripHeader';
+import {TabBarItem} from '@enonic/lib-admin-ui/ui/tab/TabBarItem';
+import {XDataWizardStepForm} from './XDataWizardStepForm';
+import {WizardStepForm} from '@enonic/lib-admin-ui/app/wizard/WizardStepForm';
+
+export class ContentWizardStepsPanel
+    extends WizardStepsPanel {
+
+    protected createHeader(header: string, wizardStepForm: WizardStepForm): PanelStripHeader {
+        const isPanelOptional = this.isWizardStepFormOptional(wizardStepForm);
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardStepsPanel.ts]:
```suggestion
        const isPanelOptional = this.isXDataStepFormOptional(wizardStepForm);
```

**Review by alansemenov — CHANGES_REQUESTED** [review]

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardStepsPanel.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    doRender(): Q.Promise<boolean> {
+        return super.doRender().then((rendered) => {
+            this.addClass('content-wizard-steps-panel');
+
+            return rendered;
+        });
+    }
+
+    private isXDataStepFormOptional(wizardStepForm: WizardStepForm): boolean {
+        return wizardStepForm instanceof XDataWizardStepForm && wizardStepForm.isExpandable();
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardStepsPanel.ts]:
```suggestion
        return wizardStepForm instanceof XDataWizardStepForm && wizardStepForm.isOptional();
```

**Review by alansemenov — CHANGES_REQUESTED** [review]

**Review by alansemenov — APPROVED** [review]

**Pull request merged**